src/md/release.md

```diff
@@ -150,6 +150,47 @@ mkdir svn-dist && cp *-source-* svn-dist/
 svn import svn-dist https://dist.apache.org/repos/dist/dev/apex/${RNAME}-RC1 -m "Apache Apex v${rv}-RC1"
 ```
 
+## Build and Deploy Documentation
+
+The documentation will be generated as static HTML files and copied into the ```apex-site``` repository. It will be available at an alternate URL location until the release is promoted, at which time it will also be promoted to the main website location.
+
+Do the following setup setups before building and deploying the documentation. Clone the apex-site repository into a folder called ```apex-site``` at the same level as the current repository. Set the following environment variables
+
+For -core releases:
+
+```
+REPO_NAME=apex-core
+DOC_NAME=apex
+```
+
+For -malhar releases:
+
+```
+REPO_NAME=apex-malhar
+DOC_NAME=malhar
+```
+
+Execute the following commands to build and deploy the documentation. **Note**: Until [mkdocs #859](https://github.com/mkdocs/mkdocs/issues/859) is resolved and available for download, use mkdocs built against [master](https://github.com/mkdocs/mkdocs).
+
+```bash
+# build docs, they would be generated in the site folder
+mkdocs build --clean
+
+# copy docs from site into target folder on apex-site
+cd ../apex-site
+git checkout asf-site
+rm -rf docs/${DOC_NAME}-${rv}
```

Comment:
@tweise Just so I understand you correctly you are suggesting option 1. below correct? If it is so, I don't mind make this change but it is technically not correct as folks who would go to the website to refer the docs will see RC docs and there could not only be mistakes in there but it could refer to features that are not yet available in the release. 

My preference is for option 3 which is what I have implemented here. I also see the additional work during promotion being minimal on what is already being done. Let me know if you still want to go with option 1. and I will make the changes as I think we have had enough discussion on this already :).

1. Publish the RC docs to the website rightaway (for example under  https://apex.apache.org/docs/apex) even though the RC hasn't been voted to be the final release and there can be other RCs after this.

2. Don't make RC docs available under any path such as https://apex.apache.org/docs/apex/3.5.0-RC1 when the RC is in review and publish them to https://apex.apache.org/docs/apex once the RC is voted to be the release.

3. What I am proposing to make RC docs available under a path such as https://apex.apache.org/docs/apex/3.5.0-RC1 while the RC is in review so that folks can review the docs as well and make it available under https://apex.apache.org/docs/apex when the release is ready.